src/app/clusters/color-control-server/color-control-server.cpp

```diff
@@ -25,11 +25,303 @@
 #include <platform/CHIPDeviceLayer.h>
 #include <platform/PlatformManager.h>
 
+#ifdef EMBER_AF_PLUGIN_SCENES
+#include <app/clusters/scenes-server/scenes-server.h>
+#endif
+
 using namespace chip;
 using namespace chip::app::Clusters;
 using namespace chip::app::Clusters::ColorControl;
 using chip::Protocols::InteractionModel::Status;
 
+#ifdef EMBER_AF_PLUGIN_SCENES
+class DefaultColorControlSceneHandler : public scenes::DefaultSceneHandlerImpl
+{
+public:
+    enum ColorControlEFS : uint8_t
+    {
+        kCurrentX = 0,
+        kCurrentY,
+        kEnhancedCurrentHue,
+        kCurrentSaturation,
+        kColorLoopActive,
+        kColorLoopDirection,
+        kColorLoopTime,
+        kColorTemperatureMireds,
+        kEnhancedColorMode,
+    };
+
+    // As per spec, 2 attributes are scenable in the level control cluster
+    static constexpr uint8_t kColorControlMaxScenableAttributes = 9;
```

Comment:
Ideally this would be somehow derived from the enum class, since it needs to stay in sync with it.

Failing that, a comment is needed here and at the end of the enum class about them needing to be in sync.

Reply:
You could add the value as the last entry of the enum.